src/adtypes_interface.jl

```diff
@@ -172,18 +172,43 @@ for detector in (:TracerSparsityDetector, :TracerLocalSparsityDetector)
 end
 
 ## Stable API to allow packages like DI to allocate caches of tracers
+
+"""
+    jacobian_tracer_type(x, detector)
+
+Return the tracer type used for Jacobian sparsity detection.
+"""
+jacobian_tracer_type(x::AbstractArray{<:Real}, ::TracerSparsityDetector{TG}) where {TG} = TG
```

Comment:
If `x` is supposed to be an `AbstractArray`, we might want to call these `jacobian_tracer_eltype` and support scalars as well?

```Julia
julia> eltype(rand(Float32, 2))
Float32

julia> eltype(0.0f0)
Float32
```

Would be a bit more intuitive to use if it mirrored a known Julia function.